server/Dockerfile

```diff
@@ -1,53 +1,39 @@
-FROM adoptopenjdk/openjdk11:x86_64-alpine-jre-11.0.3_7
+FROM adoptopenjdk/openjdk11:x86_64-alpine-jre-11.0.5_10
 
-ENV ESVERSION 6.8.0
 ENV MESH_AUTH_KEYSTORE_PATH=/keystore/keystore.jks
 ENV MESH_GRAPH_BACKUP_DIRECTORY=/backups
 ENV MESH_GRAPH_DB_DIRECTORY=/graphdb
 ENV MESH_PLUGIN_DIR=/plugins
 ENV MESH_BINARY_DIR=/uploads
+ENV MESH_TEMP_DIR=/tmp
 ENV JAVA_TOOL_OPTIONS="-Xms512m -Xmx512m -XX:MaxDirectMemorySize=256m -Dstorage.diskCache.bufferSize=256"
 
 EXPOSE 8080
 EXPOSE 8081
 
-RUN adduser -D -u 1000 -G root -h /mesh mesh
-USER mesh
-WORKDIR /mesh
-ADD ./target/mesh-server*jar /mesh/mesh.jar
-
 USER root
-RUN mkdir /graphdb   && chown mesh: /graphdb    -R && \
-    mkdir /uploads   && chown mesh: /uploads    -R && \
-    mkdir /backups   && chown mesh: /backups    -R && \
-    mkdir /plugins   && chown mesh: /plugins    -R && \
-    mkdir /keystore  && chown mesh: /keystore   -R && \
-    mkdir /config    && chown mesh: /config     -R && ln -s /config /mesh/config && \
-    mkdir /mesh/data && chown mesh: /mesh/data  -R
-
-ADD https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-$ESVERSION.tar.gz /es.tar.gz
-
-WORKDIR /
-RUN apk --update --no-cache add tar \
-    && tar xfz /es.tar.gz && apk del tar \
-    && mv elasticsearch* elasticsearch \
-    && rm /es.tar.gz \
-    && ln -s /elasticsearch /mesh/elasticsearch \
-    && mkdir -p /elasticsearch/data \
-    && chown mesh: /elasticsearch -R
-
-USER mesh
+ADD ./target/mesh-server*jar /mesh/mesh.jar
+RUN mkdir /graphdb   && chown 1000:0 /graphdb  -R && chmod 770 /graphdb  && \
+    mkdir /uploads   && chown 1000:0 /uploads  -R && chmod 770 /uploads  && \
+    mkdir /backups   && chown 1000:0 /backups  -R && chmod 770 /backups  && \
+    mkdir /plugins   && chown 1000:0 /plugins  -R && chmod 770 /plugins  && \
+    mkdir /keystore  && chown 1000:0 /keystore -R && chmod 770 /keystore && \
+    mkdir /config    && chown 1000:0 /config   -R && chmod 770 /config && ln -s /config /mesh/config && \
```

Comment:
Do the `chown` and `chmod` have any effect if the folders are overridden with mounted volumes? I thought the permissions from the host volume will be used in that case.

Reply:
No - This has only effect for non-mounted folders. When using OpenShift you usually run the pod with a random uid and mount/map config files (e.g. /config/mesh.yaml). The folder itself is however not mounted and thus writing to the folder would fail without chmod and chown.